frame/fetch.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @fileoverview Utilities for making HTTP requests to the network. */
+
+import { assertInstance } from "../lib/shared/types";
+
+/**
+ * The kinds of things that can happen as a result of trying to fetch JSON over
+ * HTTP.
+ */
+export enum FetchJsonStatus {
+  /**
+   * JSON was successfully fetched and parsed. Note that this can happen even if
+   * the response has a 4xx or 5xx status code; status codes are ignored.
```

Comment:
I think it is common for REST APIs to send 400s and 500s in JSON format in response to an illegal request. I agree that if a URL is used that's totally unprepared to serve JSON, then this will result in a potentially confusing parse error, but I think that is the normal behavior for JSON RPCs in browsers, and the parse error is still nonzero useful for debugging.

Reply:
At some point we can test Chrome's implementation and see what it does

Fine to stick with this for now though